android/src/main/java/com/pagopa/ioreactnativecrypto/X509VerificationUtil.kt

```diff
@@ -0,0 +1,300 @@
+    }
+
+    data object CRLExpired: VerificationException() {
+      private fun readResolve(): Any = CRLExpired
+    }
+  }
+
+  /**
+   * Result class for certificate chain verification
+   */
+  data class CertificateValidationResult(
+    val isValid: Boolean,
+    val validationStatus: CertificateValidationStatus,
+    val errorMessage: String? = null
+  )
+
+  /**
+   * Extracts the CRL distribution point URLs from an X509Certificate using Bouncy Castle.
+   *
+   * @param cert The X509Certificate to extract from.
+   * @return A list of CRL distribution point URLs, or an empty list if none are found.
+   */
+  private fun extractCrlDistributionPoints(cert: X509Certificate): List<String> {
+    try {
+      // Create a Bouncy Castle certificate object from the X509Certificate
+      val bcCert = JcaX509CertificateHolder(cert)
+
+      // Get the CRL distribution points extension
+      val distPoints = bcCert.getExtension(Extension.cRLDistributionPoints)
+        ?: return emptyList()
+
+      // Parse the extension value
+      val dpObj = CRLDistPoint.getInstance(distPoints.parsedValue)
+      val urls = mutableListOf<String>()
+
+      // Process each distribution point
+      for (dp in dpObj.distributionPoints) {
+        val distPointName = dp.distributionPoint
+        if (distPointName != null && distPointName.type == DistributionPointName.FULL_NAME) {
+          val generalNames = distPointName.name as GeneralNames
+          for (generalName in generalNames.names) {
+            if (generalName.tagNo == GeneralName.uniformResourceIdentifier) {
+              val uri = DERIA5String.getInstance(generalName.name).string
+              urls.add(uri)
+            }
+          }
+        }
+      }
+
+      return urls
+    } catch (e: Exception) {
+      Log.e("CertificateValidation", "Error extracting CRL distribution points", e)
+      return emptyList()
+    }
+  }
+
+  /**
+   * Downloads a CRL from a URL.
+   *
+   * @param url The URL to download the CRL from.
+   * @return The downloaded CRL as a byte array, or null if download failed.
+   */
+  private suspend fun downloadCrl(url: String): ByteArray {
+    return withContext(Dispatchers.IO) {
+      val connection = (URL(url).openConnection() as HttpURLConnection).also {
+        it.connectTimeout = 15000
+        it.readTimeout = 15000
+        it.requestMethod = "GET"
+      }
+
+      val responseCode = connection.responseCode
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        connection.inputStream.use { it.readBytes() }
+      } else {
+        throw Exception("CRL Download failed")
+      }
+    }
+  }
+
+  private suspend fun downloadCertificateCrl(
+    factory: CertificateFactory,
+    cert: X509Certificate
+  ): X509CRL {
+    val crlUrls = extractCrlDistributionPoints(cert)
+    if (crlUrls.isEmpty()) {
+      throw VerificationException.CRLNotDefined
+    }
+
+    val crlBytes = downloadCrl(crlUrls.first())
```

Comment:
I noticed we're only parsing the first CRL URL. If that URL is broken or unresponsive, we end up with a generic error. What do you think about implementing a more robust CRL handling strategy, such as trying fallback URLs or providing clearer error messages?